libs/Microsoft.MixedReality.WebRTC.Native/src/interop/global_factory.h

```diff
@@ -47,16 +82,38 @@ class GlobalFactory {
   /// object's destructor for safety.
   void RemoveObject(ObjectType type, TrackedObject* obj) noexcept;
 
+  /// Report live objects to WebRTC logging system for debugging.
+  /// This is automatically called if the |mrsShutdownOptions::kLogLiveObjects|
+  /// shutdown option is set, but can also be called manually at any time.
+  /// Return the number of live objects at the time of the call.
+  uint32_t ReportLiveObjects();
+
 #if defined(WINUWP)
   using WebRtcFactoryPtr =
       std::shared_ptr<wrapper::impl::org::webRtc::WebRtcFactory>;
   WebRtcFactoryPtr get();
   mrsResult GetOrCreateWebRtcFactory(WebRtcFactoryPtr& factory);
 #endif  // defined(WINUWP)
 
+  void AddRef() const noexcept {
+    temp_ref_count_.fetch_add(1, std::memory_order_relaxed);
+  }
+
+  void RemoveRef() const noexcept {
+    if (temp_ref_count_.fetch_sub(1, std::memory_order_acq_rel) == 1) {
+      GlobalFactory::TryShutdown();
+    }
+  }
+
  private:
-  mrsResult Initialize();
-  void ShutdownNoLock();
+  GlobalFactory(const GlobalFactory&) = delete;
+  GlobalFactory& operator=(const GlobalFactory&) = delete;
+  static std::unique_ptr<GlobalFactory>& MutableInstance(
+      bool createIfNotExist = true);
+  mrsResult InitializeImplNoLock();
+  void ForceShutdownImpl();
+  bool TryShutdownImpl();
+  void ReportLiveObjectsNoLock();
 
  private:
   rtc::scoped_refptr<webrtc::PeerConnectionFactoryInterface> factory_

```

Comment:
peerfactory_? Confusing given that the container class is also a factory

Reply:
Fixed.